Wikipedia/Code/ArticlesCollectionViewController.swift

```diff
@@ -0,0 +1,486 @@
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        collectionViewUpdater = nil
+        fetchedResultsController = nil
+        editController.close()
+    }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        cellLayoutEstimate = nil
+    }
+    
+    func configure(cell: SavedArticlesCollectionViewCell, for article: WMFArticle, at indexPath: IndexPath, layoutOnly: Bool) {
+        cell.isBatchEditing = editController.isBatchEditing
+        
+        guard let entry = readingList.entry(for: article) else {
+            return
+        }
+        
+        cell.configureAlert(for: entry, with: article, in: readingList, listLimit: dataStore.viewContext.wmf_readingListsConfigMaxListsPerUser, entryLimit: dataStore.viewContext.wmf_readingListsConfigMaxEntriesPerList.intValue)
+        cell.configure(article: article, index: indexPath.item, shouldShowSeparators: true, theme: theme, layoutOnly: layoutOnly)
+        
+        cell.isBatchEditable = true
+        cell.layoutMargins = layout.itemLayoutMargins
+        editController.configureSwipeableCell(cell, forItemAt: indexPath, layoutOnly: layoutOnly)
+    }
+    
+    func shouldDelete(_ articles: [WMFArticle], completion: @escaping (Bool) -> Void) {
+        completion(true)
+    }
+    
+    func delete(_ articles: [WMFArticle]) {
+        let url: URL? = articles.first?.url
+        let articlesCount = articles.count
+        do {
+            try dataStore.readingListsController.remove(articles: articles, readingList: readingList)
+            UIAccessibility.post(notification: UIAccessibility.Notification.announcement, argument: CommonStrings.articleDeletedNotification(articleCount: articlesCount))
+        } catch {
+            DDLogError("Error removing entries from a reading list: \(error)")
+        }
+        guard let articleURL = url, dataStore.savedPageList.entry(for: articleURL) == nil else {
+            return
+        }
+        ReadingListsFunnel.shared.logUnsaveInReadingList(articlesCount: articlesCount, language: articleURL.wmf_language)
+    }
+    
+    // MARK: - Empty state
+    
+    override func isEmptyDidChange() {
+        editController.isCollectionViewEmpty = isEmpty
+        delegate?.articlesCollectionViewControllerDidChangeEmptyState(self)
+        super.isEmptyDidChange()
+    }
+    
+    // MARK: - Refresh
+    
+    override func refresh() {
+        dataStore.readingListsController.fullSync {
+            self.endRefreshing()
+        }
+    }
+    
+    open override func endRefreshing() {
+        let now = Date()
+        let timeInterval = 0.5 - now.timeIntervalSince(refreshStart)
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + timeInterval, execute: {
+            self.collectionView.refreshControl?.endRefreshing()
+        })
```

Comment:
did you mean to add this override? looks like that's exactly what the implementation in `ColumnarCollectionViewController` is doing 😀

https://github.com/wikimedia/wikipedia-ios/blob/8be08a95ea4b11a8372e2c7eefb6589d5a8acfb0/Wikipedia/Code/ColumnarCollectionViewController.swift#L137

Reply:
Ah, great catch. I probably copied it from somewhere while refactoring and then forgot to get rid of this. Will remove this.